### DISCUSSION TITLE
Constructing spatial weights matrix for buildings based on streets

### DISCUSSION THREAD

**Discussion by mohdalattar** [pysal/momepy#453] [Q&A]
Hi,

I've calculated neighbors analysis for urban green spaces (based on first order queen contiguity) and got the following results
![image](https://user-images.githubusercontent.com/50058721/209439118-907c3844-c15d-444f-87b7-be15a4671bd5.png)

 Then, Theil Index using **area** as value of the neighboring urban green spaces and got the following:
![image](https://user-images.githubusercontent.com/50058721/209439142-6d1a4a2c-b9fe-4b74-b30f-8b2da4a83f19.png)

Not sure how to interpret the inequality results, I believe it accounts for number of neighbors and the distance ? if you could help me interpreting the two yellow urban green spaces high coefficient that would be great.

**Comment by martinfleis** [comment]
> based on first order queen contiguity

That means that nearly all polygons with a few exceptions are having 0 neighbours and the computation is meaningless. Use KNN or DistanceBand to create weights. Then it will be clearer to interpret the result.
